plugins/calibre.koplugin/metadata.lua

```diff
@@ -54,7 +54,7 @@ end
 
 -- this is the max file size we attempt to decode using json. For larger
 -- files we want to attempt to manually parse the file to avoid OOM errors
-local MAX_JSON_FILESIZE = 30 * 1000 * 1000
+local MAX_JSON_FILESIZE = 30
```

Comment:
fix me (and probably update to 50MB?)

Reply:
Maybe more like this?
https://github.com/koreader/koreader/blob/c7a59145a307b1de0e3843ffc00c541c8746999f/defaults.lua#L14-L21
https://github.com/koreader/koreader/blob/c7a59145a307b1de0e3843ffc00c541c8746999f/frontend/document/doccache.lua#L15-L21

Reply:
I don't know. I think I would like to keep a fixed size. Most of the bug reports I get are from mobileread users and seems easier for them to tell me the size of their `metadata.calibre` files than to attach logs.

But, more important than that:
- current 30MB size is too big for rapidsjon to handle in some devices/configs, for instance https://github.com/koreader/koreader/issues/11215 ***
- a sax parser, even in plain lua, should be faster than a dom parser on a scenario where we care about less than a 1-10% of the file. ****

***: most likely a setup with few books and a huge amount of user metadata per book.

****: this holds true on my tests with a little library of ~400 books and no fancy/extra metadata per book. Even after I start a wireless connection, which strips all the unused metadata from the json file, I usually got reads in the same order of magnitude using lunajson sax parser vs rapidjson dom parser.